src/adb.js

```diff
@@ -487,6 +488,220 @@ class Adb {
         );
     });
   }
+
+  // AW : return the file size of a complete folder
+  getFileSize(file) {
+    var _this = this;
+    return new Promise(function(resolve, reject) {
```

Comment:
No need to wrap this in a new promise, `getAdbState()` already returns a promise. Just return that and use `return` and `throw` in its `then()` and `catch()` blocks. See `hasAccess()` for reference.

Reply:
I tried this but the program don't wait the end of the getfilesize promise and continue it's way, the result is that I mess the display of the size on the screen. I need it to be synchrone.
`debug: Device state is device
command: adb: {"args":["-P",5037,"get-state"],"error":null,"stdout":"device\n","stderr":""}
debug: Device state is device
debug: Returned userdata FileSize is undefined Ko
command: adb: {"args":["-P",5037,"get-state"],"error":null,"stdout":"device\n","stderr":""}
debug: Device state is device
debug: Returned systemdata FileSize is undefined Ko
debug: Returned Total FileSize is NaN Ko
info: NaN
command: adb: {"args":["-P",5037,"shell","df -hBK / --output=used|tail -n1"],"error":null,"stdout":"1989568K\r\n","stderr":""}
debug: FileSize is 1989568 Ko
command: adb: {"args":["-P",5037,"shell","df -hBK /userdata --output=used|tail -n1"],"error":null,"stdout":"13062884K\r\n","stderr":""}
debug: FileSize is 13062884 Ko
`

Reply:
Then you did not do it correctly. It should work, normally. But ok, you can keep it like this for now.

Reply:
I found what I made wrong, it's now like you want.